runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -252,405 +252,258 @@ doubleMaxMinHelper(TR::Node *node, TR::CodeGenerator *cg, bool isMaxOp)
-   regDeps->addPostCondition(rTgtBase      ,TR::RealRegister::AssignAny);
-   regDeps->addPostCondition(rTgtVal       ,TR::RealRegister::AssignAny);
-   regDeps->addPostCondition(rSrcValIdx    ,TR::RealRegister::AssignAny);
-
-   regDeps->addPostCondition(rLen          ,TR::RealRegister::AssignAny);
-   regDeps->addPostCondition(rResidue      ,TR::RealRegister::AssignAny);
-
-   regDeps->addPostCondition(vTmp          ,TR::RealRegister::AssignAny);
-   regDeps->addPostCondition(vCaseBuf      ,TR::RealRegister::AssignAny);
-   regDeps->addPostCondition(vBuf          ,TR::RealRegister::AssignAny);
-
-   regDeps->addPostCondition(vOffset       ,TR::RealRegister::AssignAny);
-   regDeps->addPostCondition(vAlphaRange   ,TR::RealRegister::AssignAny);
-   regDeps->addPostCondition(vAlphaCntrl   ,TR::RealRegister::AssignAny);
-
-   regDeps->addPostCondition(vInvalidRange ,TR::RealRegister::AssignAny);
-   regDeps->addPostCondition(vInvalidCntrl ,TR::RealRegister::AssignAny);
-
-   if (cg->supportsHighWordFacility() &&
-           !comp->getOption(TR_DisableHighWordRA) &&
-           TR::Compiler->target.is64Bit())
-      {
-      rResidue->setIs64BitReg(true);
-      rLen->setIs64BitReg(true);
-      rSrcValIdx->setIs64BitReg(true);
-      rTgtVal->setIs64BitReg(true);
-      rSrcVal->setIs64BitReg(true);
-      }
-
-   // byte [] java.lang.String.value   the raw byte array
-   // int     java.lang.String.count   the count i.e num of characters. A Unicode point is 1 character. Surrogate's are 2 characters.
-   // valueConentOffset                the offset of actual content in a char array.
-   int32_t stringValOffset, stringCountOffset, valueContentOffset;
-
-   // TODO (Filip): This is a workaround for Java 829 performance as we switched to using a byte[] backing array in String*. Remove this workaround once obsolete.
-   TR_OpaqueClassBlock *stringClass = cg->fej9()->getClassFromSignature("Ljava/lang/String;", strlen("Ljava/lang/String;"), comp->getCurrentMethod(), true);
-
-   if (cg->fej9()->getInstanceFieldOffset(stringClass, "value", "[B") != ~0)
-      {
-      stringValOffset = cg->fej9()->getInstanceFieldOffsetIncludingHeader("Ljava/lang/String;", "value", "[B", comp->getCurrentMethod());
+  /** \brief
+   *     Attempts to use vector registers to perform SIMD conversion of characters from lowercase to uppercase
+   *
+   *  \detail
+   *     Uses vector registers to convert 16 bytes at a time
+   *
+   *  \param node
+   *     The node representing the HW optimized toUpper and toLower recognized calls
+   *
+   *  \param cg
+   *     The code generator used to generate the instructions
+   *
+   *  \param isToUpper
+   *     Boolean representing case conversion, either to upper or to lower
+   *
+   *  \param isCompressedString
+   *     Boolean representing the string's compression
+   *
+   *  \return
+   *     0 if the conversion is unsuccessful, a non zero value otherwise
```

Comment:
This is not true. The function returns a register which represents the return value of the Java call. We need to JavaDoc the JIT helpers you added to define what the input parameters represent and what the return value represents. This Doxygen comment needs to be modified.

Reply:
Added JavaDoc comments to JIT helper methods and fixed doxygen comment in 91004aa